From ad029f029c339f82cd6a32eb9c31075ee506715e Mon Sep 17 00:00:00 2001
From: Lukasz Bartosik <lbartosik@marvell.com>
Date: Fri, 1 Feb 2019 19:34:50 +0530
Subject: [PATCH 030/767] octeontx2-af: don't register CPT AF interrupts

commit f2b1be2e7d7b596766a6f8fc9145361fac2d0afe from
git@git.assembla.com:cavium/WindRiver.linux.git

Don't register CPT AF interrupts if CPT block is not
implemented on a given platform.

Change-Id: I3fcf69f2cae14979eb3037e8391789800d74f34e
Signed-off-by: Lukasz Bartosik <lbartosik@marvell.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index 9e1195f54297..ed4bef545624 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -2582,9 +2582,11 @@ static int rvu_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	if (err)
 		goto err_flr;
 
-	err = rvu_cpt_register_interrupts(rvu);
-	if (err)
-		goto err_irq;
+	if (is_block_implemented(rvu->hw, BLKADDR_CPT0)) {
+		err = rvu_cpt_register_interrupts(rvu);
+		if (err)
+			goto err_irq;
+	}
 
 	err = rvu_policy_init(rvu);
 	if (err)
-- 
2.31.1

